### PR TITLE
Issue/navigation

### PR DESCRIPTION
**Pull request by kevvo15**
Addressed issues regarding the navigation logic of my application. (Issue: #13) Some changes include:

- `ActivityIndicator` to inform users that content is loading.
- Back Tab navigation fixes
- Drawer content no longer overlaps with status bar

### CODE REVIEW

File: screens/Signup.js [hunk 1]
```diff
@@ -19,17 +19,21 @@ const registerSchema = yup.object({
     .required()
 });
 
+
+
 export default function Signup({ navigation }) {
 
    // Email and Password User Creation
    createUser = (email, password) => {
```
Comment (CTOverton) [inline screens/Signup.js:27]:
```suggestion
   const createUser = (email, password) => {
```

**Review by CTOverton — APPROVED** [review]

**Comment by mukesh525** [comment]
@CTOverton  it seems @kevvo15 has done all suggested changes. 
 To learn more about navigation it will be good if he explore navigation states ,tab navigator and passing data using props and most important **redux** flow(store ,provider, thunk) .

**Review by mukesh525 — APPROVED** [review]

File: routes/homeStack.js [hunk 2]
```diff
@@ -15,23 +15,33 @@ import SongDetails from '../screens/SongDetails';
...
         }
     },
     Home: {
         screen: Home,
-        navigationOptions: {
-            title: 'HOME'
+        navigationOptions: ({ navigation }) => {
+            return {
+                headerLeft: ()=> null,
+                headerTitle: () => <Header navigation={navigation} title="MUSIC BASE" />
```
Comment (mukeshjha2207) [inline routes/homeStack.js:39]: @kevvo15  try using header has custom reusable  component.

**Review by mukeshjha2207 — APPROVED** [review]

File: screens/Home.js [hunk 3]
```diff
@@ -11,13 +12,24 @@ export default function Home({ navigation }) {
         { section: 'ARTISTS', key: '3'}
     ])
 
+    // Signs the user out
+  const signOut = () => {
+    auth()
+    .signOut()
+    .then(() => {
+        console.log("User successfully signed out.");
+        navigation.replace("Login");
```
[the remark this answers is not in the document]
Comment (kevvo15) [reply screens/Home.js:21]: @CTOverton thanks for the info! I will have to look at ways to implement this `navigation.reset` as the current suggestion does not seem to work as intended. I tried the format provided in the documentation which also seems to not work as expected. This is something I'll continue to research before implementation :)

File: screens/Login.js [hunk 4]
```diff
@@ -50,24 +53,27 @@ export default function Login({ navigation }) {
...
       auth()
       .signOut()
-      .then(() => console.log("User successfully signed out."));
+      .then(() => console.log("User successfully signed out.")); //navigation.reset and navigate to login page
   }
 
   // Handle user state changes
   function onAuthStateChanged(user) {
     setUser(user);
-    if (initializing) setInitializing(false);
```
[the remark this answers is not in the document]
Comment (kevvo15) [reply screens/Login.js:62]: Ah I see what you mean here! Yes thank you, I did intend to skip it all together but I appreciate the extra info!

**Pull request merged**